services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/helpers/Wrappers.groovy

```diff
@@ -34,33 +35,38 @@ class Wrappers {
     }
 
     /**
-     * Wait for closure to return a result. Returns false if success condition was not met.
+     * Wait for closure to return a result.
      *
      * @param timeout time to wait in seconds
      * @param retryInterval pause between retries (in seconds)
      * @param closure : A closure that returns following:
-     *          - True if check is successful
-     *          - False or null if check is unsuccessful
-     * @return True if wait was successful, false otherwise
+     *          - Groovy truth or null if check is successful
+     *          - False or throw if check is unsuccessful
+     * @return True if wait was successful, throws WaitTimeoutException otherwise
      */
     static boolean wait(double timeout, double retryInterval = 0.5, Closure closure) {
         long endTime = System.currentTimeMillis() + (long) (timeout * 1000)
         long sleepTime = (long) (retryInterval * 1000)
+        def thrown
         while (System.currentTimeMillis() < endTime) {
             try {
-                if (closure.call()) {
+                def result = closure.call()
+                if (result || result == null) {
                     return true
                 } else {
-                    log.debug("No wait results yet. Sleeping for ${sleepTime} ms...")
+                    thrown = null
                     sleep(sleepTime)
                 }
             } catch (Throwable t) {
-                log.debug("Wait check threw exception: ${t.message}.\nSleeping for ${sleepTime} ms...")
+                thrown = t
                 sleep(sleepTime)
             }
         }
-        log.debug("Closure failed to succeed within ${timeout} seconds, aborting.")
-        return false
+        def message = "Condition didn't pass in $timeout seconds"
```

Comment:
I would prefer such a variant:

Condition was not satisfied within ${timeout} seconds